src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -810,14 +814,30 @@ private void GetCodeForExternMethodAccessor(ExternMethodAccessorNode node)
             }
 
             LLVMModuleRef externFuncModule = _moduleWithExternalFunctions;
-            Debug.Assert(externFuncModule.GetNamedFunction(externFuncName).Handle == IntPtr.Zero);
-            LLVMValueRef externFunc = externFuncModule.AddFunction(externFuncName, externFuncType);
+            WasmImportKey wasmImportKey = new WasmImportKey(node.ExternSymbolKey);
+            bool externExists = _wasmImportLinkages.TryGetValue(wasmImportKey, out LLVMValueRef externFunc);
```

Comment:
This I am struggling to understand the need for `_wasmImportLinkages`. The node factory should unique accessor nodes for the same WASM import, so why would we be adding the same one twice?

Reply:
You are right, it can all be done the same as direct P/Invokes.